framework/service/org/moqui/impl/InstanceServices.xml

```diff
@@ -214,8 +214,13 @@ along with this software (see the LICENSE.md file). If not, see
             <entity-find-one entity-name="moqui.server.instance.AppInstance" value-field="appInstance"/>
             <set field="hostType" from="appInstance?.instanceHost?.hostType"/>
             <if condition="hostType == null"><return error="true" message="No InstanceHostType found for AppInstance with ID ${appInstanceId}"/></if>
-            <service-call name="${hostType.initService}" in-map="context" out-map="context"/>
-        </actions>
+            <!-- <service-call name="${hostType.initService}" in-map="context" out-map="context"/>-->
+            <!--For async call -->
+            <script><![CDATA[
+            import org.moqui.context.ExecutionContext
+            import java.util.concurrent.Future
+            Future<Map> instanceFuture = ec.service.async().name("${hostType.initService}").parameters([context:context,appInstanceId:appInstanceId]).callFuture()
```

Comment:
What's the point of having calling this service async?

Reply:
The point of making this service **async** is to prevent user/client from getting stuck when the image is not available. In synchronous calls, the user would have to wait for the image to be pulled, which could take 6-8 minutes depending on the size. By using asynchronous calls, the user can request to start the container, and if the image is not available, it will be automatically pulled in the background while the user can continue with other tasks. This improves user experience and efficiency.